pyacq/viewers/qtimefreq.py

```diff
@@ -0,0 +1,739 @@
+from pyqtgraph.Qt import QtCore, QtGui
+import pyqtgraph as pg
+from pyqtgraph.util.mutex import Mutex
+import vispy.color
+
+import sys
+import numpy as np
+import weakref
+import time
+from collections import OrderedDict
+
+from ..core import (WidgetNode, Node, register_node_type,  InputStream, OutputStream,
+        ThreadPollInput, StreamConverter, StreamSplitter)
+
+from .qoscilloscope import MyViewBox
+
+try:
+    import scipy.signal
+    import scipy.fftpack
+    HAVE_SCIPY = True
+except ImportError:
+    HAVE_SCIPY = False
+
+
+default_params = [
+        {'name': 'xsize', 'type': 'float', 'value': 10., 'step': 0.1, 'limits' : (.1, 60)},
+        {'name': 'nb_column', 'type': 'int', 'value': 1},
+        {'name': 'background_color', 'type': 'color', 'value': 'k' },
+        #~ {'name': 'colormap', 'type': 'list', 'value': 'hot', 'values' : ['hot', 'coolwarm', 'ice', 'grays', ] },
+        {'name': 'colormap', 'type': 'list', 'value': 'hot', 'values' : list(vispy.color.get_colormaps().keys()) },
+        {'name': 'refresh_interval', 'type': 'int', 'value': 500 , 'limits':[5, 1000]},
+        {'name': 'mode', 'type': 'list', 'value': 'scroll' , 'values' : ['scan', 'scroll'] },
+        {'name': 'show_axis', 'type': 'bool', 'value': False},
+        #~ {'name': 'display_labels', 'type': 'bool', 'value': False }, #TODO when title
+        {'name' : 'timefreq' , 'type' : 'group', 'children' : [
+                        {'name': 'f_start', 'type': 'float', 'value': 3., 'step': 1.},
+                        {'name': 'f_stop', 'type': 'float', 'value': 90., 'step': 1.},
+                        {'name': 'deltafreq', 'type': 'float', 'value': 3., 'step': 1.,  'limits' : [0.1, 1.e6]},
+                        {'name': 'f0', 'type': 'float', 'value': 2.5, 'step': 0.1},
+                        {'name': 'normalisation', 'type': 'float', 'value': 0., 'step': 0.1},]}
+    ]
+
+default_by_channel_params = [ 
+                {'name': 'visible', 'type': 'bool', 'value': True},
+                {'name': 'clim', 'type': 'float', 'value': 1.},
+            ]
+
+
+class QTimeFreq(WidgetNode):
+    """
+    Class to visulaize time-frequency with Morlet continuous wavelet.
+    
+    It allow better visualisation the standart FFT spectrogram because it is not dependant
+    of the FFT size along frequency.
+    See https://en.wikipedia.org/wiki/Morlet_wavelet
+    
+    The implementation is done with one TimeFreqWorker per channel.
+    The computation is quite heavy:
+        * downsampling the signal chunk (with a filtfilt first)
+        * fft of signal
+        * multiplication line by line (freq by freq) with the FFT wavelet.
+        * ifft the maps
+        
+    There is 2 modes:
+        * each TimeFreqWorker live in the same QApp as QTimeFreq (nodegroup_friends=None)
+        * each TimeFreqWorker is spawn in some other NodeGroup to dustrubute the  (nodegroup_friends=[some_list_of_nodegroup])
```

Comment:
Thanks but at the moment. We have limitation due to RPC. this second mode imply that QTimeFreq is in local QApp and connot be remoted because nodegroup_friends cannot be Jsonified.
But TimeFreqWorker can be on another machine and this is amazing. 64 channels at 20kHz 2FPS and no lag!!!